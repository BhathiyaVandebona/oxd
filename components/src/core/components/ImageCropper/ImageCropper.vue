<template>
  <cropper
    class="cropper"
    :src="image"
    :stencil-component="$options.components.CircleStencil"
    :default-size="{
      width: stencilHeight,
      height: stencilHeight,
    }"
    :default-position="defaultPosition"
    :min-height="stencilHeight"
    :min-width="stencilHeight"
    @change="change"
  />
</template>

<script lang="ts">
import {defineComponent, ref} from 'vue';
import {Cropper, CircleStencil} from 'vue-advanced-cropper';
export default defineComponent({
  name: 'ImageCropper',
  components: {
    Cropper,
    CircleStencil,
  },
  props: {
    image: {
      type: String,
    },
    defaultPositionLeft: {
      type: String,
      default: '500',
    },
    defaultPositionTop: {
      type: String,
      default: '500',
    },
    stencilWidth: {
      type: String,
      default: '50',
    },
    stencilHeight: {
      type: String,
      default: '50',
    },
  },
  emits: ['change'],

  setup(props, {emit}) {
    const croppedImage = ref('');
    const change = ({
      coordinates,
      canvas,
    }: {
      coordinates: string;
      canvas: HTMLCanvasElement;
    }) => {
      croppedImage.value = canvas.toDataURL();
      emit('change', croppedImage.value);
    };

    const defaultPosition = () => {
      return {
        left: props.defaultPositionLeft,
        top: props.defaultPositionTop,
      };
    };
    return {
      croppedImage,
      change,
      defaultPosition,
    };
  },
});
</script>
<style src="./image-cropper.scss" lang="scss" scoped></style>
