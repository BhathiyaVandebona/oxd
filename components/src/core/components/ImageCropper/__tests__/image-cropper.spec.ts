import {shallowMount} from '@vue/test-utils';
import ImageCropper from '@orangehrm/oxd/core/components/ImageCropper/ImageCropper.vue';

describe('ImageCropper.vue', () => {
  it('renders ImageCropper', () => {
    const wrapper = shallowMount(ImageCropper, {
      props: {
        image:
          'https://images.unsplash.com/photo-1485178575877-1a13bf489dfe?ixlib=rb-1.2.1&auto=format&fit=crop&w=991&q=80',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
