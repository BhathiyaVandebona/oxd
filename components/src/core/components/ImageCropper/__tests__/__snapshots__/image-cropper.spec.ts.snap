// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCropper.vue renders ImageCropper 1`] = `
<cropper-stub defaultsize="[object Object]" defaultposition="function () {
      return {
        left: props.defaultPositionLeft,
        top: props.defaultPositionTop
      };
    }" minheight="50" minwidth="50" src="https://images.unsplash.com/photo-1485178575877-1a13bf489dfe?ixlib=rb-1.2.1&amp;auto=format&amp;fit=crop&amp;w=991&amp;q=80" stencilcomponent="[object Object]" backgroundwrappercomponent="[object Object]" stencilprops="[object Object]" autozoom="false" debounce="500" transitions="true" checkorientation="true" canvas="true" transitiontime="300" imagerestriction="fit-area" roundresult="true" priority="coordinates" resizeimage="true" moveimage="true" resizealgorithm="function(t){var e=t.event,i=t.coordinates,n=t.aspectRatio,o=t.positionRestrictions,s=t.sizeRestrictions,r=p(p({},i),{right:i.left+i.width,bottom:i.top+i.height}),a=e.params||{},h=p({},e.directions),c=a.allowedDirections||{left:!0,right:!0,bottom:!0,top:!0};s.widthFrozen&amp;&amp;(h.left=0,h.right=0),s.heightFrozen&amp;&amp;(h.top=0,h.bottom=0),j.forEach((function(t){c[t]||(h[t]=0)}));var l=V(r,h=St({coordinates:r,directions:h,sizeRestrictions:s,positionRestrictions:o})).width,u=V(r,h).height,d=a.preserveRatio?G(r):J(l/u,n);if(d){var m=a.respectDirection;if(m||(m=r.width>=r.height||1===d?&quot;width&quot;:&quot;height&quot;),&quot;width&quot;===m){var g=l/d-r.height;if(c.top&amp;&amp;c.bottom){var f=h.top,v=h.bottom;h.bottom=Mt(g,v,f),h.top=Mt(g,f,v)}else c.bottom?h.bottom=g:c.top?h.top=g:c.right?h.right=0:c.left&amp;&amp;(h.left=0)}else if(&quot;height&quot;===m){var b=r.width-u*d;if(c.left&amp;&amp;c.right){var w=h.left,y=h.right;h.left=-Mt(b,w,y),h.right=-Mt(b,y,w)}else c.left?h.left=-b:c.right?h.right=-b:c.top?h.top=0:c.bottom&amp;&amp;(h.bottom=0)}h=St({directions:h,coordinates:r,sizeRestrictions:s,positionRestrictions:o,preserveRatio:!0,compensate:a.compensate})}return l=V(r,h).width,u=V(r,h).height,(d=a.preserveRatio?G(r):J(l/u,n))&amp;&amp;Math.abs(d-l/u)>.001&amp;&amp;j.forEach((function(t){c[t]||(h[t]=0)})),dt({event:new E({left:-h.left,top:-h.top}),coordinates:{width:i.width+h.right+h.left,height:i.height+h.top+h.bottom,left:i.left,top:i.top},positionRestrictions:o})}" movealgorithm="function dt(t){var e=t.event,i=t.coordinates,n=t.positionRestrictions,o=void 0===n?{}:n,s=Z(i,e.directions);return Z(s,K(s,o))}" initstretcher="function(t){var e=t.stretcher,i=t.imageSize,n=G(i);e.style.width=i.width+&quot;px&quot;,e.style.height=e.clientWidth/n+&quot;px&quot;,e.style.width=e.clientWidth+&quot;px&quot;}" fitcoordinates="function(t){var e=t.visibleArea,i=t.coordinates,n=t.aspectRatio,o=t.sizeRestrictions,s=t.positionRestrictions,r=p(p({},i),ut({width:i.width,height:i.height,aspectRatio:n,sizeRestrictions:{maxWidth:e.width,maxHeight:e.height,minHeight:Math.min(e.height,o.minHeight),minWidth:Math.min(e.width,o.minWidth)}}));return r=ot(r=Z(r,Y(F(i),F(r))),ht(X(e),s))}" fitvisiblearea="function(t){var e=t.visibleArea,i=t.boundaries,n=t.getAreaRestrictions,o=t.coordinates,s=p({},e);s.height=s.width/G(i),s.top+=(e.height-s.height)/2,(o.height-s.height>0||o.width-s.width>0)&amp;&amp;(s=_(s,Math.max(o.height/s.height,o.width/s.width)));var r=q(K(o,X(s=_(s,nt(s,n({visibleArea:s,type:&quot;resize&quot;}))))));return s.width<o.width&amp;&amp;(r.left=0),s.height<o.height&amp;&amp;(r.top=0),s=ot(s=Z(s,r),n({visibleArea:s,type:&quot;move&quot;}))}" arearestrictionsalgorithm="function(t){var e=t.visibleArea,i=t.boundaries,n=t.imageSize,o=t.imageRestriction,s=t.type,r={};return&quot;fill-area&quot;===o?r={left:0,top:0,right:n.width,bottom:n.height}:&quot;fit-area&quot;===o&amp;&amp;(G(i)>G(n)?(r={top:0,bottom:n.height},e&amp;&amp;&quot;move&quot;===s&amp;&amp;(e.width>n.width?(r.left=-(e.width-n.width)/2,r.right=n.width-r.left):(r.left=0,r.right=n.width))):(r={left:0,right:n.width},e&amp;&amp;&quot;move&quot;===s&amp;&amp;(e.height>n.height?(r.top=-(e.height-n.height)/2,r.bottom=n.height-r.top):(r.top=0,r.bottom=n.height)))),r}" sizerestrictionsalgorithm="function(t){return{minWidth:t.minWidth,minHeight:t.minHeight,maxWidth:t.maxWidth,maxHeight:t.maxHeight}}" positionrestrictionsalgorithm="function(t){var e=t.imageSize,i={};return&quot;none&quot;!==t.imageRestriction&amp;&amp;(i={left:0,top:0,right:e.width,bottom:e.height}),i}" class="cropper"></cropper-stub>
`;
